### PR TITLE
Add activity permission

### PR DESCRIPTION
**Pull request by Yelinz**
Remove unnecessary activity field in reports and make activity read only if it has been transferred.

### CODE REVIEW

File: timed/tracking/models.py [hunk 1]
```diff
@@ -82,11 +82,6 @@ class Report(models.Model):
     review       = models.BooleanField(default=False)
     not_billable = models.BooleanField(default=False)
     task         = models.ForeignKey('projects.Task', related_name='reports')
-    activity     = models.ForeignKey(Activity,
```
[the remark this answers is not in the document]
Comment (Yelinz) [reply timed/tracking/models.py]: Opened #309

File: timed/tracking/views.py [hunk 2]
```diff
@@ -23,6 +23,10 @@ class ActivityViewSet(ModelViewSet):
 
     serializer_class = serializers.ActivitySerializer
     filterset_class     = filters.ActivityFilterSet
+    permission_classes = [
+        # users may not change transferred activities
+        C(IsAuthenticated) & C(IsTransferred)
```
[the remark this answers is not in the document]
Comment (Yelinz) [reply timed/tracking/views.py]: Made an get test with transferred true.

File: timed/permissions.py [hunk 3]
```diff
@@ -91,3 +91,10 @@ def has_permission(self, request, view):
 
     def has_object_permission(self, request, view, obj):
         return self.has_permission(request, view)
+
+
+class IsTransferred(BasePermission):
+    """Allows access only to not transferred objects."""
+
+    def has_object_permission(self, request, view, obj):
+        return obj.transferred is False
```
[the remark this answers is not in the document]
Comment (Yelinz) [reply timed/permissions.py]: Done

File: timed/tracking/tests/test_activity.py [hunk 4]
```diff
@@ -213,3 +213,36 @@ def test_activity_to_before_from(auth_client):
...
+
+    url = reverse('activity-detail', args=[activity.id])
+    response = auth_client.patch(url, data)
+    assert response.status_code == status.HTTP_403_FORBIDDEN
+
+
+def test_activity_retrievable_not_editable(auth_client):
+    """Test that transferred activities are still retrievable."""
+    activity = ActivityFactory.create(user=auth_client.user, transferred=True)
+    url = reverse('activity-list')
```
Comment (sliverc) [inline timed/tracking/tests/test_activity.py]: I am more worried that -detail doesn't work...

Comment (Yelinz) [reply timed/tracking/tests/test_activity.py]: Added a read only permission and changed the test to -detail

**Comment by sliverc** [comment]
:heart:

**Pull request merged**